### PR TITLE
PEN-957: Add School Master Read API

### PR DESCRIPTION
**Pull request by andyyang**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/school/repository/SchoolRepository.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import ca.bc.gov.educ.api.school.model.SchoolEntity;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * The interface Pen name text repository.
+ */
+@Repository
+public interface SchoolRepository extends CrudRepository<SchoolEntity, MinCode> {
+  //Optional<SchoolEntity> findStudentEntityByPen(String pen);
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/school/repository/SchoolRepository.java:13]: can this commented out code be removed?

File: api/src/main/java/ca/bc/gov/educ/api/school/config/SchoolAPIMVCConfig.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * The type School api mvc config.
+ *
+ * @author Om
+ */
+@Configuration
+public class SchoolAPIMVCConfig implements WebMvcConfigurer {
+
+  /**
+   * The Pen reg api interceptor.
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/school/config/SchoolAPIMVCConfig.java:19]: Some comments are referring to different api in some of these files

File: api/src/main/java/ca/bc/gov/educ/api/school/service/SchoolService.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  /**
+   * Search for SchoolEntity by MinCode
+   *
+   * @param minCode the unique minCode for a given school.
+   * @return the School entity if found.
+   */
+  public SchoolEntity retrieveSchoolByMinCode(String minCode) {
+    Optional<SchoolEntity> result = Optional.empty();
+    if(minCode.length() > DISTNO_LENGTH) {
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/school/service/SchoolService.java:43]: Does minCode need a null check?

File: tools/jenkins/Jenkinsfile-api [hunk 4]
```diff
@@ -0,0 +1,107 @@
+pipeline{
```
Comment (mightycox) [inline tools/jenkins/Jenkinsfile-api:1]: Do we need jenkins file for openshift 3?

File: api/src/main/java/ca/bc/gov/educ/api/school/repository/SchoolRepository.java (hunk 1 above)
Comment (andyyang) [reply api/src/main/java/ca/bc/gov/educ/api/school/repository/SchoolRepository.java:13]: Let me delete it

File: api/src/main/java/ca/bc/gov/educ/api/school/config/SchoolAPIMVCConfig.java (hunk 2 above)
Comment (andyyang) [reply api/src/main/java/ca/bc/gov/educ/api/school/config/SchoolAPIMVCConfig.java:19]: Let me delete it

File: api/src/main/java/ca/bc/gov/educ/api/school/service/SchoolService.java (hunk 3 above)
Comment (andyyang) [reply api/src/main/java/ca/bc/gov/educ/api/school/service/SchoolService.java:43]: minCode is required by the controller, so it should not be null.

File: tools/jenkins/Jenkinsfile-api (hunk 4 above)
Comment (andyyang) [reply tools/jenkins/Jenkinsfile-api:1]: Let me delete the file.

**Pull request merged**